container_api/src/container_admin.rs

```diff
@@ -0,0 +1,615 @@
+        let mut container = Container::from_config(config.clone());
+        container.dna_loader = test_dna_loader();
+        container.load_config().unwrap();
+
+        let mut tmp_config_path = PathBuf::new();
+        tmp_config_path.push(format!("./tmp-{}-container-config.toml", test_name.into()));
+        container.set_config_path(tmp_config_path.clone());
+        container
+    }
+
+    #[test]
+    fn test_install_dna_from_file() {
+        let mut container = create_test_container("test_install_dna_from_file");
+
+        let mut new_dna_path = PathBuf::new();
+        new_dna_path.push("new-dna.hcpkg");
+
+        assert_eq!(
+            container.install_dna_from_file(new_dna_path.clone(), String::from("new-dna")),
+            Ok(()),
+        );
+
+        let new_dna =
+            Arc::get_mut(&mut test_dna_loader()).unwrap()(&String::from("new-dna.hcpkg")).unwrap();
+
+        assert_eq!(container.config().dnas.len(), 2,);
+        assert_eq!(
+            container.config().dnas,
+            vec![
+                DnaConfiguration {
+                    id: String::from("test-dna"),
+                    file: String::from("app_spec.hcpkg"),
+                    hash: String::from("Qm328wyq38924y"),
+                },
+                DnaConfiguration {
+                    id: String::from("new-dna"),
+                    file: String::from("new-dna.hcpkg"),
+                    hash: String::from(new_dna.address()),
+                },
+            ]
+        );
+
+        let mut config_contents = String::new();
+        let mut file = File::open(&container.config_path).expect("Could not open temp config file");
+        file.read_to_string(&mut config_contents)
+            .expect("Could not read temp config file");
+
+        assert_eq!(
+            config_contents,
+r#"bridges = []
+
+[[agents]]
+id = "test-agent-1"
+key_file = "holo_tester.key"
+name = "Holo Tester 1"
+public_address = "HoloTester1-----------------------------------------------------------------------AAACZp4xHB"
+
+[[agents]]
+id = "test-agent-2"
+key_file = "holo_tester.key"
+name = "Holo Tester 2"
+public_address = "HoloTester2-----------------------------------------------------------------------AAAGy4WW9e"
+
+[[dnas]]
+file = "app_spec.hcpkg"
+hash = "Qm328wyq38924y"
+id = "test-dna"
+
+[[dnas]]
+file = "new-dna.hcpkg"
+hash = "QmPB7PJUjwj6zap7jB7oyk616sCRSSnNFRNouqhit6kMTr"
+id = "new-dna"
+
+[[instances]]
+agent = "test-agent-1"
+dna = "test-dna"
+id = "test-instance-1"
+
+[instances.storage]
+type = "memory"
+
+[[instances]]
+agent = "test-agent-2"
+dna = "test-dna"
+id = "test-instance-2"
+
+[instances.storage]
+type = "memory"
+
+[[interfaces]]
+admin = true
+id = "websocket interface"
+
+[[interfaces.instances]]
+id = "test-instance-1"
+
+[[interfaces.instances]]
+id = "test-instance-2"
+
+[interfaces.driver]
+port = 3000
+type = "websocket"
+
+[logger]
+type = ""
+[[logger.rules.rules]]
```

Comment:
This needs to be here because the defaults get added when saving the config. So we need to add it to the fixture. I will post my 2nd PR on ContainerAdmin in a few minutes that includes a DRY out of these fixtures..

Reply:
@zippy @lucksus In my scenario-tests branch I make "debug" the default logger_type because I assume that's what we want, would you agree?

Reply:
Yeah probably but this indicates that we need to think through different default conditions.  For running scenario tests, and end user probably wants the default logging just to show `debug/dna` messages, and maybe `err/*`  For the container, the default is likely no debug messages.   For us as holochain core devs, the default is what it currently is:  All messages.